pkg/inject/oauth_proxy.go

```diff
@@ -20,23 +20,59 @@ const defaultProxySecret = "ncNDoqLGrayxXzxTn5ANbOXZp3qXd0LA"
 
 // OAuthProxy injects an appropriate proxy into the given deployment
 func OAuthProxy(jaeger *v1.Jaeger, dep *appsv1.Deployment) *appsv1.Deployment {
-	if jaeger.Spec.Ingress.Security != v1.IngressSecurityOAuthProxy {
-		return dep
+	if jaeger.Spec.Ingress.Security == v1.IngressSecurityOAuthProxy {
+	    dep.Spec.Template.Spec.ServiceAccountName = account.OAuthProxyAccountNameFor(jaeger)
+
+	    if viper.GetString("platform") == v1.FlagPlatformOpenShift {
+            dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, getOpenShiftOAuthProxyContainer(jaeger))
+            dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, corev1.Volume{
+                Name: service.GetTLSSecretNameForQueryService(jaeger),
+                VolumeSource: corev1.VolumeSource{
+                    Secret: &corev1.SecretVolumeSource{
+                        SecretName: service.GetTLSSecretNameForQueryService(jaeger),
+                    },
+                },
+            })
+	    } else {
+            dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, getOAuthProxyContainer(jaeger))
+	    }
 	}
-
-	dep.Spec.Template.Spec.ServiceAccountName = account.OAuthProxyAccountNameFor(jaeger)
-	dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, getOAuthProxyContainer(jaeger))
-	dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, corev1.Volume{
-		Name: service.GetTLSSecretNameForQueryService(jaeger),
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName: service.GetTLSSecretNameForQueryService(jaeger),
-			},
-		},
-	})
 	return dep
 }
 
+func getOAuthProxyContainer(jaeger *v1.Jaeger) corev1.Container {
```

Comment:
One thing I just realized while reviewing this PR: we want the generic OAuth feature to allow users to specify their own OAuth Proxies, so that we are not tied to Keycloak specifically.

With that in mind, the first consumer that overrides the default OAuth Proxy could be the operator itself, when running on OpenShift. Meaning: the code path for the generic OAuth proxy could be used for both OpenShift and non-OpenShift, with the OpenShift path only customizing the OAuth parts (image/args).

I know it might sound a bit frustrating to hear this at this point in the PR, but it just clicked with me. For some reason, I believe you'll also realize that :)